src/memory_server.ml

```diff
@@ -25,6 +25,26 @@ open Xapi_stdext_unix
 module D = Debug.Make(struct let name = Memory_interface.service_name end)
 open D
 
+module Host_free_memory_event = struct
+  open Xapi_stdext_threads
+  let accesible = ref false
+  let c = Condition.create ()
+  let m = Mutex.create ()
+
+  let wait () =
+    Threadext.Mutex.execute m (fun () ->
```

Comment:
You could take a look at the Event module, it might allow you to implement what you need simpler by using Event.sent and Event.receive

Reply:
Yes, I have seen that module, but it does not suitable for this case because we may call the api will finally call the `Event.receive` multiple times, but only call the api which finally calls `Event.send`  once.